### ISSUE TITLE
False positives in socket library

### ISSUE THREAD

**Issue by nodeg** [PyCQA/pylint#4798]
### Bug description

Pylint complains about non existent members in the socket library of Python, although they exist according to the [docu](https://docs.python.org/3.8/library/socket.html).

Sample code:
```python
#!/usr/bin/python3

"""
socket_test.py
Pylint test for the socket library
"""
import socket


HOSTNAME = "suse.com"
IPADDR = ""

try:
    IPADDR = socket.gethostbyname_ex(HOSTNAME)
except socket.gaierror:
    IPADDR = "10.0.0.1"
print(IPADDR)
```

### Command used

```shell
pylint socket_test.py
```

### Pylint output

```shell
************* Module socket_test
socket_test.py:13:13: E1101: Module 'socket' has no 'gethostbyname_ex' member (no-member)
socket_test.py:14:7: E1101: Module 'socket' has no 'gaierror' member (no-member)

---------------------------------------------------------------------
Your code has been rated at -4.29/10 (previous run: -12.86/10, +8.57)
```

### Expected behavior

Pylint should pass without error.

### Pylint version

```shell
pylint 2.8.3
astroid 2.5.6
Python 3.8.10 (default, May 05 2021, 15:36:36) [GCC]
```

### OS / Environment

openSUSE Tumbleweed

**Comment by Pierre-Sassoulas** [comment]
I don't have any problem with the latest pylint, what happen if you use 2.9.6 or the latest master ?

**Comment by nodeg** [comment]
I updated pylint with pipx and still got the same issue:

```bash
$ pipx upgrade-all
(...)              
upgraded package pylint from 2.8.3 to 2.9.6 (location: /home/dom/.local/pipx/venvs/pylint)
(...)

$ pylint --version
pylint 2.9.6
astroid 2.6.6
Python 3.8.10 (default, May 05 2021, 15:36:36) [GCC]

$ pylint socket_test.py
************* Module socket_test
socket_test.py:13:13: E1101: Module 'socket' has no 'gethostbyname_ex' member (no-member)
socket_test.py:14:7: E1101: Module 'socket' has no 'gaierror' member (no-member)

--------------------------------------------------------------------
Your code has been rated at -2.50/10 (previous run: -2.50/10, +0.00)
```

The master branch did not work for me either:

```bash
$ pylint --version
pylint 2.10.0-dev0
astroid 2.6.6
Python 3.8.10 (default, May 05 2021, 15:36:36) [GCC]

$ pylint socket_test.py
************* Module socket_test
socket_test.py:13:13: E1101: Module 'socket' has no 'gethostbyname_ex' member (no-member)
socket_test.py:14:7: E1101: Module 'socket' has no 'gaierror' member (no-member)

------------------------------------
Your code has been rated at -2.50/10
```

Any suggestions?

**Comment by Pierre-Sassoulas** [comment]
Could you provide your list of dependencies ? Or if you don't have one the result of `pip freeze` ? :)

**Comment by nodeg** [comment]
I installed `pylint` via pipx and not via pip.
```bash
$ pip freeze
appdirs==1.4.4
attrs==21.2.0
chardet==4.0.0
cli-test-helpers==1.0.1
iniconfig==1.1.1
ordered-set==4.0.2
packaging==21.0
pep517==0.11.0
pluggy==0.13.1
psutil==5.8.0
py==1.10.0
pycurl==7.43.0.6
PyGObject==3.40.1
pyparsing==2.4.7
pytest==6.2.4
six==1.16.0
toml==0.10.2
tomli==1.2.0
wfuzz==3.1.0

$ pipx list
venvs are in /home/dom/.local/pipx/venvs
apps are exposed on your $PATH at /home/dom/.local/bin
   package bpytop 1.0.67, Python 3.8.10
    - bpytop
   package dirsearch 0.4.2, Python 3.8.10
    - dirsearch
   package pylint 2.9.6, Python 3.8.10
    - epylint
    - pylint
    - pyreverse
    - symilar
   package sqlmap 1.5.8, Python 3.8.10
    - sqlmap
   package virtualenv 20.7.0, Python 3.8.10
    - virtualenv
```

**Comment by Pierre-Sassoulas** [comment]
Still can't reproduce with:
```
pylint 2.9.6
astroid 2.6.6
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0]
```
Using the same code and same name, after ``sudo apt install libcurl4-openssl-dev libssl-dev libcairo2-dev libgirepository1.0-dev``, upgrading pip and installing latest wheel + installing the dependencies from your ``pip freeze``.

**Comment by DudeNr33** [comment]
This looks very similar to the problem discussed in #4759 - some modules implemented in C give false-positive ``no-member`` or similar errors for ``socket`` and ``math`` (and possibly also for other C modules in the standard library). 

In this case it depended on the way Python itself has been installed, but I did not figure out yet what the core problem is.

**Comment by nodeg** [comment]
Thanks for the comment @DudeNr33. I can confirm the issue on macOS using Homebrew, too.

```bash
$ pylint --version
pylint 2.9.6
astroid 2.6.6
Python 3.9.6 (default, Jun 29 2021, 05:25:02) 
[Clang 12.0.5 (clang-1205.0.22.9)]

$ pylint socket_test.py
************* Module socket_test
socket_test.py:14:13: E1101: Module 'socket' has no 'gethostbyname_ex' member (no-member)
socket_test.py:15:7: E1101: Module 'socket' has no 'gaierror' member (no-member)

------------------------------------
Your code has been rated at -2.50/10
```

**Comment by DudeNr33** [comment]
You can try to use ``pyenv`` to manage the Python installations for you - this helped me and another user to get rid of those false positives.
It seems like ``pyenv`` compiles CPython from source, so doing that yourself might be another option (until a true fix for this issue is found).

**Comment by nodeg** [comment]
Thanks for the suggestions. I have no problems with the pylint result since I know that those socket methods do exist. I just wanted to report these false positives.
